lms/djangoapps/teams/views.py

```diff
@@ -830,6 +837,7 @@ def get(self, request):
                     CourseKey.from_string(course_key_string)
                     for course_list in [enrolled_courses, staff_courses]
                     for course_key_string in course_list
+                    if course_id is None or course_key_string == course_id
```

Comment:
While this may be currently working, I'm finding this really difficult to read, as the logic is nested inside a couple levels of a list comprehension.

I think it'd be best to keep all the course-checking logic as close together as possible, since these API methods tend to grow pretty long. I recommend: if `course_id` is passed in 1) validate it is a real course id, 2) validate that the team_id denotes a team belonging to the course_id, and then finally 3) restricting `valid_courses` to only contain the specified course.